### DISCUSSION TITLE
Are Avalonia Material's main colors really restricted to the PrimaryColor & SecondaryColor lists? Why?

### DISCUSSION THREAD

**Discussion by buzzware** [AvaloniaCommunity/Material.Avalonia#368] [Q&A] [answered]
I tried to to set colors using a corporate theme, then discovered :
1) MaterialTheme PrimaryColor & SecondaryColor properties are enums with a short list of Material colors. It appears you can't set just any RGB color
2) I can't even use any defined Material color, like  
```<themes:MaterialTheme PrimaryColor="Indigo900" SecondaryColor="LightGreen500" />```

Is there no override? Is this intentional? What is the reasoning ?

**Comment by SKProCH** [comment]
Hello. 
Currently there is an Advanced Theming wiki page that describes how you can fully customize Material theme: https://github.com/AvaloniaCommunity/Material.Avalonia/wiki/Advanced-Theming

**Comment by Keboo** [comment]
I ran into this recently as well, and rolled a `ColorTheme` that allows arbitrary color for the primary and secondary.
It works very similar to [this](https://github.com/MaterialDesignInXAML/MaterialDesignInXamlToolkit/blob/eb9338c720dd93fd53aef6d6499586efa63a87ca/src/MainDemo.Wpf/App.xaml#L21).

Happy to open a PR if there is any interest in adding it.

**Reply by SKProCH** [reply]
Hmm, is the `CustomColorTheme` contains just colors (without importing styles/resources)? If yes, that's the point of having separated `CustomColorTheme` if colors already provided by `BundledTheme`?

**Reply by Keboo** [reply]
For my use case (and I suspect the same as the original question), is I have a company color palate that I need to implement that does not match any of the existing material design colors. 

The limitation with `BundleTheme`/`MaterialTheme` is that it only allows you to set the Primary and Secondary from the pre-defined set of the colors in the enum. This is convenient when you only want to use the Material Design colors, but does not work when you have an existing color palate that you want to use instead. The idea of having a second derived class of `MaterialThemeBase` that allows setting arbitrary colors is it allows for easy setting up of the theme colors without a bunch of extra work to take over the entirety of the `ITheme`.  Happy to open a PR if you would like to see an implementation of what this could look like.

**Reply by SKProCH** [reply]
Hmm, this makes sense actually. Yep, PR would be welcomed. 

Btw, as far as I remember - you are the maintainer of MDiX, how did it happen that you came to Avalonia?)

**Reply by Keboo** [reply]
Great I will see about getting something together for it.

Yes I am the maintainer of MDiX. Been doing some Avalonia work in my day job so was poking around at what Avalonia projects were out there. Was really happy to see a Material implementation like this.

**Reply by Keboo** [reply]
Sorry for the delay (life got busy), I just opened PR #413 with a proposed solution.

**Comment by Keboo** [comment, marked as answer]
Just closing the loop on this. Version 3.8.0 added a new CustomMaterialTheme that can be used for this. 

Details in the wiki: https://github.com/AvaloniaCommunity/Material.Avalonia/wiki/Advanced-Theming#custommaterialtheme
